src/planner/kinodynamic/KinodynamicTimer.cpp

```diff
@@ -0,0 +1,273 @@
+#include "aikido/planner/kinodynamic/KinodynamicTimer.hpp"
+#include <dart/dart.hpp>
+#include <aikido/common/Spline.hpp>
+#include <aikido/common/StepSequence.hpp>
+
+#include "Path.h"
+#include "Trajectory.h"
+
+namespace aikido {
+namespace planner {
+namespace kinodynamic {
+
+//==============================================================================
+std::unique_ptr<aikido::trajectory::Spline>
+createSplineFromWaypointsAndConstraints(
+    const std::list<Eigen::VectorXd>& waypoints,
+    const Eigen::VectorXd& maxVelocities,
+    const Eigen::VectorXd& maxAccelerations,
+    aikido::statespace::ConstStateSpacePtr stateSpace,
+    double maxDeviation,
+    double timeStep,
+    double startTime)
+{
+  Trajectory trajectory(
+      Path(waypoints, maxDeviation), maxVelocities, maxAccelerations, timeStep);
+  if (trajectory.isValid())
+  {
+    using dart::common::make_unique;
+    using CubicSplineProblem = aikido::common::
+        SplineProblem<double, int, 4, Eigen::Dynamic, Eigen::Dynamic>;
+
+    std::size_t dimension = stateSpace->getDimension();
+    double endTime = startTime + trajectory.getDuration();
+
+    // create spline
+    auto outputTrajectory
+        = make_unique<aikido::trajectory::Spline>(stateSpace, startTime);
+
+    // create a sequence of time steps from start time to end time by time step
+    aikido::common::StepSequence sequence(
+        timeStep, true, true, startTime, endTime);
+
+    const Eigen::VectorXd zeroPosition = Eigen::VectorXd::Zero(dimension);
+    auto currState = stateSpace->createState();
+    for (std::size_t i = 1; i < sequence.getLength(); i++)
```

Comment:
I guess this is a personal preference, but can we start `i = 0` and go to `sequence.getLength() - 1`?